src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -221,6 +223,18 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             {
                 return HttpContextExpr;
             }
+            else if (parameter.ParameterType == typeof(HttpRequest))
+            {
+                return HttpRequestExpr;
+            }
+            else if (parameter.ParameterType == typeof(HttpResponse))
+            {
+                return HttpResponseExpr;
+            }
+            else if (parameter.ParameterType == typeof(ClaimsPrincipal))
+            {
+                return UserExpr;
+            }
```

Comment:
I wasn't sure what the best sequencing might be for the conditions here for if there was an optimal order from a perf perspective, so this order is kinda arbitrary.

Reply:
It doesn't *really* matter since this happens once at startup.